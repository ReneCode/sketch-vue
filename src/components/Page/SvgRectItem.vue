<template>
  <rect :x="item.svg.x" :y="item.svg.y" :width="item.svg.width" :height="item.svg.height" :style="style">
  </rect>
</template>

<script>
export default {
  name: 'svg-rect-item',
  props: ['item'],
  computed: {
    style() {
      let style = {
      };
      if (this.item.svg.stroke) {
        style.stroke = this.item.svg.stroke;
      }
      if (this.item.svg.strokeWidth) {
        style.strokeWidth = this.item.svg.strokeWidth;
      }
      if (this.item.svg.fill) {
        style.fill = this.item.svg.fill;
      }
      return style;
    }
  }
}
</script>

<style>
rect {
  fill: none;
}
</style>
