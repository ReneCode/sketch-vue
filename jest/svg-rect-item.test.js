import { mount } from 'vue-test-utils'
import SvgRectItem from '../src/components/Page/SvgRectItem'
import ItemRectangle from '../src/models/item-rectangle'
import Point from '../src/models/point'

describe('SvgRectItem', () => {

  const p1 = new Point(33, 44);
  const p2 = new Point(55, 77);
  const itemRect = new ItemRectangle(p1, p2);
  itemRect.svg.stroke = "red";
  itemRect.svg.strokeWidth = "3px";
  itemRect.svg.fill = "blue";

  const wrapper = mount(SvgRectItem, {
    propsData: {
      item: itemRect
    }
  })

  it('renders the markup', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })
})
