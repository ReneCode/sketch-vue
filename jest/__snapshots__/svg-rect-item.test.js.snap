// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgRectItem renders the markup 1`] = `<rect x="33" y="44" width="22" height="33" style="stroke: red; stroke-width: 3px; fill: blue;"></rect>`;
